### PR TITLE
I/O error handling

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/solver/writer/immediate_file_writer.cpp [hunk 1]
```diff
@@ -39,35 +57,49 @@ ImmediateFileResultWriter::ImmediateFileResultWriter(const char* folderOutput) :
...
-static Yuni::String prepareDirectoryHierarchy(const YString& root, const std::string& entryPath)
+static bool prepareDirectoryHierarchy(const YString& root, const std::string& entryPath, Yuni::String& output)
 {
-    Yuni::String output;
     output << root << Yuni::IO::Separator << entryPath.c_str();
-    createDirectoryHierarchy(root, entryPath.c_str());
-    return output;
+    if (!createDirectoryHierarchy(root, entryPath.c_str()))
+        return false;
+    return true;
```
Comment (flomnes) [inline src/solver/writer/immediate_file_writer.cpp]:
```suggestion
    return createDirectoryHierarchy(root, entryPath.c_str());
```

File: src/solver/writer/immediate_file_writer.cpp [hunk 2]
```diff
@@ -39,35 +57,47 @@ ImmediateFileResultWriter::ImmediateFileResultWriter(const char* folderOutput) :
...
-    IOFileSetContent(output, entryContent);
+    Yuni::String output;
+    if (!prepareDirectoryHierarchy(pOutputFolder, entryPath, output))
+    {
+        return;
+    }
+    else
+    {
+        IOFileSetContent(output, entryContent);
+    }
```
Comment (flomnes) [inline src/solver/writer/immediate_file_writer.cpp]:
```suggestion
    if (prepareDirectoryHierarchy(pOutputFolder, entryPath, output))
    {
        IOFileSetContent(output, entryContent);
    }
```

File: src/solver/writer/zip_writer.cpp [hunk 3]
```diff
@@ -31,24 +31,41 @@ ZipWriteJob<ContentT>::ZipWriteJob(ZipWriter& writer,
...
+    info.compression_method = MZ_COMPRESS_METHOD_DEFLATE;
+    info.modified_date = info.creation_date = std::time(0);
+}
+
 template<class ContentT>
 void ZipWriteJob<ContentT>::onExecute()
 {
+    int32_t bw, ret;
     mz_zip_file file_info;
+    fillInfo(file_info, pEntryPath);
```
Comment (guilpier-code) [inline src/solver/writer/zip_writer.cpp]: Why not : 
```C++
mz_zip_file file_info = setInfo(file_info, pEntryPath);
```
?

File: src/solver/writer/zip_writer.cpp [hunk 4]
```diff
@@ -62,14 +79,23 @@ ZipWriter::ZipWriter(std::shared_ptr<Yuni::Job::QueueService> qs,
...
+        logs.error() << "Error opening zip file " << pArchivePath << " (" << ret << ")";
+    }
     // TODO : make level of compression configurable
     mz_zip_writer_set_compress_level(pZipHandle, MZ_COMPRESS_LEVEL_FAST);
 }
 
 ZipWriter::~ZipWriter()
 {
-    mz_zip_writer_close(pZipHandle);
+    int ret;
```
Comment (guilpier-code) [inline src/solver/writer/zip_writer.cpp]: why not directly : 
```C++
int ret = mz_zip_writer_close(pZipHandle);
```
?

**Review by guilpier-code — CHANGES_REQUESTED** [review]

**Pull request merged**